lshell/checkconfig.py

```diff
@@ -575,7 +576,11 @@ def get_config_user(self):
             self.conf['history_file'] = "%s/%s" % (self.conf['home_path'],
                                                    self.conf['history_file'])
 
-        os.environ['PATH'] = os.environ['PATH'] + self.conf['env_path']
+        # add env_path at the beginning of the PATH env variable if specified
+        if self.conf['env_path_before_os_path'] is 1:
+            os.environ['PATH'] = self.conf['env_path'] + os.environ['PATH']
```

Comment:
Do you see any reason why we shouldn't just replace the current implementation with your line: `os.environ['PATH'] = self.conf['env_path'] + os.environ['PATH']` ?

What I mean here, is that we would not add a new configuration flag, but just have this new `$PATH` behavior by default.

Reply:
I didn't want to change the behavior of your code so I chose to add a parameter and have its default not changing the order of `env_path` . Now I think that because there is a case (e.g. rbenv) where one needs to have `env_path` is at the beginning of the `PATH` env var, then there surely must also be a case where a user will require to have `env_path` added at the end of `PATH` (the default). In practice you usually have `/usr/bin:/bin:...` at the beginning of your `PATH` env variable and custom/user pathes at the end,  so actually your default is fine. To cut a long story short I think it would be nice to keep this extra parameter I added just in case.

Regarding rbenv I managed to make it work. It looks like the `env_path` order was the last piece missing in the puzzle.